### PR TITLE
Progress boxes overhaul

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
Rewrote the progress boxes so that the loop is in game.component, as makes more sense. The code certainly looks better than it did before, although getting it to work felt rather a Pyrrhic victory in the end.

Added default visibility to the lines on the charts, although this would probably be better as an optional rather than mandatory argument. Tweaked the maximum calculation so it behaves a bit better, although it could still be further improved for the large disparity in MM's assets chart.

Also implemented auto-numbering of boxes in CSS, which ended up being subject to Angular's non-cascading stylesheets, but does at least work. Indeed, with my knowledge of javascript, this is the only way to do it, and seems to me to be considerably more elegant than some kind of injective solution.

### CODE REVIEW

File: src/app/common-components/n64-box/n64-box.component.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <section>
   <ng-container *ngIf="!simple">
     <header>
-      <div class="box slot">File {{ number }}</div>
+      <div class="box slot"><div class="filenum">File </div></div>
```
Comment (rozzan) [inline src/app/common-components/n64-box/n64-box.component.html:4]: Does this work cross component? i.e. If you have two game-progress components and a game-faq component, does this track correctly?

**Review by rozzan — COMMENTED** [review]
Just one thing to confirm!

File: src/app/common-components/n64-box/n64-box.component.html (hunk 1 above)
Comment (EllipticEllipsis) [reply src/app/common-components/n64-box/n64-box.component.html:4]: image
Yes sir

Comment (EllipticEllipsis) [reply src/app/common-components/n64-box/n64-box.component.html:4]: image

**Pull request merged**